### PR TITLE
some typing and code tweaks in MZV

### PR DESCRIPTION
**Pull request by fchapoton**
### :books: Description

- Add some basic typing annotations.
- Try to gain speed by using `_monomial` method.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/modular/multiple_zeta.py [hunk 1]
```diff
@@ -248,7 +250,7 @@ def coproduct_iterator(paire):
                                            tail[step:]))
 
 
-def composition_to_iterated(w, reverse=False):
+def composition_to_iterated(w, reverse=False) -> tuple[int, ...]:
```
Comment (mkoeppe) [inline src/sage/modular/multiple_zeta.py:253]: This annotation requires Python >= 3.9, but we still support 3.8;
see https://docs.python.org/3/library/typing.html#typing.Tuple

File: src/sage/modular/multiple_zeta.py [hunk 2]
```diff
@@ -380,7 +382,7 @@ def minimize_term(w, cf):
 
 # numerical values
 
-class MultizetaValues(UniqueRepresentation):
+class MultizetaValues(Singleton):
```
Comment (mkoeppe) [inline src/sage/modular/multiple_zeta.py:385]: Does this pickle correctly (does it have to pickle)?

File: src/sage/modular/multiple_zeta.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/modular/multiple_zeta.py:253]: even with the added line `from __future__  import annotations` ?

File: src/sage/modular/multiple_zeta.py (hunk 2 above)
Comment (fchapoton) [reply src/sage/modular/multiple_zeta.py:385]: It does not make much sense to pickle that.

File: src/sage/modular/multiple_zeta.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/modular/multiple_zeta.py:253]: We already have the exact same annotation in `src/sage/combinat/interval_posets.py` and other places.

Comment (mkoeppe) [reply src/sage/modular/multiple_zeta.py:253]: OK, thanks.

**Pull request merged**